### DISCUSSION TITLE
Certificate-based authentication for the LDAP service account

### DISCUSSION THREAD

**Discussion by ivanov17** [authelia/authelia#5215] [Q&A] [answered]
Hello, I have [reported this as a bug](https://github.com/authelia/authelia/issues/5207), but it has been closed. I was informed that this should be a support question.

Some things are not well documented, so I'm a bit confused about the  capabilities of password reset feature.

I can specify `authentication_backend.ldap.tls.certificate_chain` and `authentication_backend.ldap.tls.private_key` options in the Authelia configuration file (or set them using secrets) to secure LDAP connections.

As I expect, this should allow me to use the same certificate/key pair with SASL EXTERNAL mechanism to authenticate a user that is allowed lookup and change password operations. In short, I should be able to use certificate-based authentication.

But Authelia requires credentials for the LDAP service account. 

I can omit `authentication_backend.ldap.user` and `authentication_backend.ldap.password` options only if I set `authentication_backend.ldap.permit_unauthenticated_bind` option to `true`.

But I can't set `authentication_backend.ldap.permit_unauthenticated_bind` option to `true` if I set `false` for `authentication_backend.password_reset.disable`. 

Currently I must to provide service account credentials if I want to use user password reset feature.

I thought that the problem is that Authelia forbids password reset with unauthenticated bind, whereas SASL authentication, including the EXTERNAL mechanism, should be handled without an LDAP binding.

But maybe some magic with configuration options is needed if this is a support question?

**Comment by james-d-elliott** [comment]
I think what you meant to say is:

> I couldn't find an answer to this question in the docs and I appreciate the time you take in answering this query. 
> The configuration docs for LDAP list TLS configurations and allow specifying a certificate chain and key. Does this mean you intended support SASL External Bind or is there something I'm missing? When I configure this in the below example configuration it seems to also require a username and password which doesn't seem right to me.

**Reply by james-d-elliott** [reply, marked as answer]
> I couldn't find an answer to this question in the docs and I appreciate the time you take in answering this query.
 > The configuration docs for LDAP list TLS configurations and allow specifying a certificate chain and key. Does this mean you intended support SASL External Bind or is there something I'm missing? When I configure this in the below example configuration it seems to also require a username and password which doesn't seem right to me.

Hey @ivanov17 thanks for your polite and respectful query. We don't support and have not claimed to support SASL External Bind at this time unfortunately. You can always make a feature request for this feature. You can also open a pull request to clarify this in the docs in a way that would help users like yourself in the future.
